WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel.swift

```diff
@@ -23,6 +24,10 @@ final class PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel: Observabl
     func onAppear() {
         showsInfoSheet = true
     }
+
+    static func == (lhs: PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel, rhs: PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel) -> Bool {
+        lhs.id == rhs.id
+    }
```

Comment:
We should implement equatable fully here – identity comparison can lead to issues in future when testing, as we assume that the return would change if any of the underlying properties changed.

I know it's not possible for them to change or differ between instances at the moment, but this implementation isn't resillient to changes in the view model.

As a minimum, we could add a test counting the number of properties (see other view models e.g. `CardPresentPaymentsModalButtonViewModelTests`) to remind ourselves to update the == function if the number of properties changes... but it would be better to also include the other properties, in case they are changed to `var`, or can be passed in with different values through the `init`.

Reply:
> As a minimum, we could add a test counting the number of properties (see other view models e.g. CardPresentPaymentsModalButtonViewModelTests)

`XCTAssertPropertyCount` is super clever. I have never used it myself! 👍  

I'll add these updates in a next PR.

Reply:
> I have never used it myself! 👍
I only added it last week :)